src/UI/Art/SpriteArtist/UnitSpriteSet.java

```diff
@@ -206,12 +206,13 @@ public void drawUnit(Graphics g, AnimState state, int imageIndex, int drawX, int
     // Draw the unit, facing the appropriate direction.
     if( flipImage && isStateFlippable(state) )
     {
-      g2d.drawImage(frame, drawX - shiftX + (frame.getWidth()), drawY + shiftY, -frame.getWidth(), frame.getHeight(), null);
+      int flippedDrawX = drawX + SpriteLibrary.baseSpriteSize;
+      g2d.drawImage(frame, flippedDrawX - shiftX, drawY + shiftY, -frame.getWidth(), frame.getHeight(), null);
```

Comment:
Huh. Looking at the initialization of `shiftX`, it appears that it will always be zero, and does not match the comment ("center X") because the enforced order of operations is wrong.

That said - it looks like `flippedDrawX` assumes that sprites have the standard width. Otherwise shouldn't it be `drawX + SpriteLibrary.baseSpriteSize + ((frame.getWidth()-SpriteLibrary.baseSpriteSize) / 2)`? i.e., draw from the right bound for normal-sized sprites, and shift others by half the difference between normal and actual sprite width?

Reply:
Maybe I should just make up some irregular sprites and try it to see.

Reply:
Ah, okay. I don't know why I didn't think of the wider movement sprites, or the fact that I was reinventing the wheel I was complaining about. Sigh.